src/main/java/uk/gov/hmcts/reform/sscs/service/docmosis/PdfLetterService.java

```diff
@@ -0,0 +1,91 @@
+        this.docmosisPdfService = docmosisPdfService;
+        this.docmosisTemplatesConfig = docmosisTemplatesConfig;
+    }
+
+    public byte[] generateCoversheet(NotificationWrapper wrapper, SubscriptionType subscriptionType) {
+        Address addressToUse = getAddressToUseForLetter(wrapper, subscriptionType);
+        PdfCoverSheet pdfCoverSheet = new PdfCoverSheet(
+                wrapper.getCaseId(),
+                addressToUse.getLine1(),
+                addressToUse.getLine2(),
+                addressToUse.getTown(),
+                addressToUse.getCounty(),
+                addressToUse.getPostcode()
+        );
+        String templatePath = docmosisTemplatesConfig.getCoversheets().get(wrapper.getNotificationType().getId());
+        if (StringUtils.isBlank(templatePath)) {
+            throw new PdfGenerationException(
+                    String.format("There is no template for notificationType %s",
+                            wrapper.getNotificationType().getId()),
+                    new RuntimeException("Invalid notification type for docmosis coversheet."));
+        }
+        return docmosisPdfService.createPdf(pdfCoverSheet, templatePath);
+    }
+
+
+    public byte[] generateLetter(NotificationWrapper wrapper, Notification notification, SubscriptionType subscriptionType) {
+        try {
+            if (NotificationEventType.APPEAL_RECEIVED_NOTIFICATION.getId().equalsIgnoreCase(wrapper.getNotificationType().getId())) {
+
+
+                Address addressToUse = getAddressToUseForLetter(wrapper, subscriptionType);
+                Map<String, String> placeholders = notification.getPlaceholders();
+                placeholders.put(SSCS_URL_LITERAL, SSCS_URL);
+                placeholders.put(GENERATED_DATE_LITERAL, LocalDateTime.now().toLocalDate().toString());
+                placeholders.put(ADDRESS_LINE_1, addressToUse.getLine1());
+                placeholders.put(ADDRESS_LINE_2, addressToUse.getLine2() == null ? " " : addressToUse.getLine2());
+                placeholders.put(ADDRESS_LINE_3, addressToUse.getTown() == null ? " " : addressToUse.getTown());
+                placeholders.put(ADDRESS_LINE_4, addressToUse.getCounty() == null ? " " : addressToUse.getCounty());
+                placeholders.put(POSTCODE_LITERAL, addressToUse.getPostcode());
+                placeholders.put(docmosisTemplatesConfig.getHmctsImgKey1(), docmosisTemplatesConfig.getHmctsImgVal());
+                byte[] letter = docmosisPdfService.createPdf(placeholders, notification.getDocmosisLetterTemplate());
+
+                byte[] coversheet = LetterUtils.addBlankPageAtTheEndIfOddPage(generateCoversheet(wrapper, subscriptionType));
+                return LetterUtils.buildBundledLetter(LetterUtils.buildBundledLetter(letter, coversheet), coversheet);
+            }
+        } catch (IOException e) {
+            log.error("Cannot generate letter", e);
```

Comment:
Output the notification event type here to make it easier to identify what type of letter failed.